miniapp/miniapp_triangular_solver.cpp

```diff
@@ -0,0 +1,210 @@
+  MatrixType b(GlobalElementSize{opts.m, opts.n}, TileElementSize{opts.mb, opts.nb}, comm_grid);
+
+  const auto side = blas::Side::Left;
+  const auto uplo = blas::Uplo::Lower;
+  const auto op = blas::Op::NoTrans;
+  const auto diag = blas::Diag::NonUnit;
+  const T alpha = 2.0;
+
+  double m = A.size().rows();
+  double n = b.size().cols();
+  auto add_mul = n * m * m / 2;
+  const double total_ops = dlaf::total_ops<T>(add_mul, add_mul);
+
+  using dlaf::matrix::test::getLeftTriangularSystem;
+  std::function<T(const GlobalElementIndex&)> setter_A, setter_b, expected_b;
+  std::tie(setter_A, setter_b, expected_b) =
+      getLeftTriangularSystem<GlobalElementIndex, T>(uplo, op, diag, alpha, A.size().rows());
+
+  for (auto run_index = 0; run_index < opts.nruns; ++run_index) {
+    if (0 == world.rank())
+      std::cout << "[" << run_index << "]" << std::endl;
+
+    // setup matrix A and b
+    using dlaf::matrix::util::set;
+    set(A, setter_A);
+    set(b, setter_b);
+
+    // wait all setup tasks before starting benchmark
+    ::waitall_tiles(A);
+    ::waitall_tiles(b);
+    MPI_Barrier(world);
+
+    dlaf::common::Timer<> timeit;
+    dlaf::Solver<Backend::MC>::triangular(comm_grid, side, uplo, op, diag, alpha, A, b);
+
+    // wait for last task and barrier for all ranks
+    ::waitall_tiles(A);
+    ::waitall_tiles(b);
+    MPI_Barrier(world);
+
+    auto elapsed_time = timeit.elapsed();
+
+    // compute gigaflops
+    double gigaflops = total_ops / elapsed_time / 1e9;
+
+    // print benchmark results
+    if (0 == world.rank())
+      std::cout << "[" << run_index << "]"
+                << " " << elapsed_time << "s"
+                << " " << gigaflops << "GFlop/s"
+                << " " << A.size() << " " << A.blockSize() << " " << comm_grid.size() << " " << b.size()
+                << " " << b.blockSize() << " " << hpx::get_os_thread_count() << std::endl;
+
+    // (optional) run test
+    if (opts.do_check) {
+      // TODO evaluate to change check
+      CHECK_MATRIX_NEAR(expected_b, b, 20 * (b.size().rows() + 1) * TypeUtilities<T>::error,
+                        20 * (b.size().rows() + 1) * TypeUtilities<T>::error);
+    }
+  }
+
+  return hpx::finalize();
+}
+
+int main(int argc, char** argv) {
+  dlaf::comm::mpi_init mpi_initter(argc, argv, dlaf::comm::mpi_thread_level::serialized);
+
+  // options
+  using namespace hpx::program_options;
+  options_description desc_commandline("Usage: " HPX_APPLICATION_STRING " [options]");
+
+  // clang-format off
+  desc_commandline.add_options()
+    ("matrix-size",       value<SizeType>()->default_value(4096),  "Matrix size.")
+    ("block-size",        value<SizeType>()->default_value(256),   "Block cyclic distribution size.")
+    ("result-cols",       value<SizeType>()->default_value(512),   "Matrix size.")
+    ("result-block-cols", value<SizeType>()->default_value(512),   "Block cyclic distribution size.")
```

Comment:
What about adding the matrix the values referred to? For example "Matrix A size" and "Matrix b size", or "Input matrix size" and "Output matrix size".

Reply:
I renamed the parameters according to the `trsm` function, so all the sizes refers to the B matrix, from which it is possible to infer all other sizes.